library/core/src/option.rs

```diff
@@ -1064,9 +1066,13 @@ impl<T: Copy> Option<&T> {
     /// let copied = opt_x.copied();
     /// assert_eq!(copied, Some(12));
     /// ```
+    #[rustc_const_unstable(feature = "const_option", issue = "67441")]
     #[stable(feature = "copied", since = "1.35.0")]
-    pub fn copied(self) -> Option<T> {
-        self.map(|&t| t)
+    pub const fn copied(self) -> Option<T> {
+        match self {
```

Comment:
Same thing here, it looks like the implementation at https://doc.rust-lang.org/src/core/option.rs.html#486 was pasted in here and slightly modified. I don't think this is good. Could you add a `TODO` like above to change this back to `self.map(|&t| t)` once possible?

Reply:
Added a FIXME for this then. It does sound like a better idea than repeating what amounts to the same piece of code in multiple places.